### PR TITLE
[wip] ask about detailing surface=paved/unpaved

### LINKED ISSUE #279: add quest to make surface=paved/unpaved more detailed
Currently surface quest for detailed surface info, but there is no quest to make general surface values more clear.

And there is significant difference between surface=cobblestone and surface=asphalt, surface=grass_paver or surface=compacted (especially for cases like people on wheelchairs or cyclists).

I propose to add similar quests, asked for surface=paved and surface=unpaved and asking to make these data more detailed - it will allow both to make existing data more detailed and stop discouraging answering general answer.

For example I just cycled several streets, all paved. I would solve these quests as I am now waiting in queue - but as I am not remembering which part was surface=paving_stones and answering surface=paved would hide theses quests I decided to not do anything.

----

query: exclude ways with `cycleway:surface`, `surface:cycleway`, `footway:surface`, `surface:footway`, `segregated=yes`

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #279

### CODE REVIEW

**Comment by matkoniecz** [comment]
It may make sense to add more surface values if users frequently create notes that some exotic surface value must be added.

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurfaceForm.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+public class DetailPavedRoadSurfaceForm extends RoadSurfaceForm {
+	RoadSurfaceForm.Surface[] GetSurfaceMenuStructure()
+	{
+		return new RoadSurfaceForm.Surface[]{
+				new RoadSurfaceForm.Surface("asphalt", R.drawable.surface_asphalt, R.string.quest_surface_value_asphalt),
+				new RoadSurfaceForm.Surface("concrete", R.drawable.surface_concrete, R.string.quest_surface_value_concrete),
+				new RoadSurfaceForm.Surface("sett", R.drawable.surface_sett, R.string.quest_surface_value_sett),
+				new RoadSurfaceForm.Surface("paving_stones", R.drawable.surface_paving_stones, R.string.quest_surface_value_paving_stones),
+				new RoadSurfaceForm.Surface("cobblestone", R.drawable.surface_cobblestone, R.string.quest_surface_value_cobblestone),
+				new RoadSurfaceForm.Surface("wood", R.drawable.surface_wood, R.string.quest_surface_value_wood),
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurfaceForm.java]: The "paved road surface select" is a simple selection like the roofs and sports, not a grouped select, it should use the simpler component as well.

I think for many cases in software project where I at one point thought inheritance is considered harmful was where higher order of complexity was introduced in base classes and then concealed/not used in child classes.

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	protected String getTagFilters()
+	{
+		return " ways with ( highway ~ " + TextUtils.join("|", RoadSurfaceConfig.ROADS_WITH_SURFACES) + " and" +
+				" surface=paved)";
+	}
+
+	@Override
+	public int importance()
+	{
+		return QuestImportance.MINOR;
```
Comment (matkoniecz) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java]: EXTRA? INSIGNIFICANT is not really fitting as surface data is used.

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	@Inject public DetailPavedRoadSurface(OverpassMapDataDao overpassServer)
+	{
+		super(overpassServer);
+	}
+
+	@Override
+	protected String getTagFilters()
+	{
+		return " ways with ( highway ~ " + TextUtils.join("|", RoadSurfaceConfig.ROADS_WITH_SURFACES) + " and" +
+				" surface=paved)";
```
Comment (matkoniecz) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java]: done

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java (hunk 2 above)
Comment (matkoniecz) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/DetailPavedRoadSurface.java]: downgraded to EXTRA

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/RoadSurfaceConfig.java [hunk 4]
```diff
@@ -10,4 +10,12 @@
 			"track", "road",
 			/*"service", */ // this is too much, and the information value is very low
 	};
+
+	static final String[] ALL_ROADS = {
+			"trunk","trunk_link","motorway","motorway_link",
+			"primary", "primary_link", "secondary", "secondary_link", "tertiary", "tertiary_link",
+			"unclassified", "residential", "bicycle_road", "living_street", "pedestrian",
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/RoadSurfaceConfig.java]: highway=bicycle_road does not exists.

Comment (matkoniecz) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/RoadSurfaceConfig.java]: https://github.com/westnordost/StreetComplete/issues/332 (it is also in unmodified SC)

Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/RoadSurfaceConfig.java]: Ah, you moved the code. Opened PR.

Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/RoadSurfaceConfig.java]: Fixing it here could be the easiest solution.

**Comment by westnordost** [comment]
Fyi, I also added a quest icon for this quest.

**Comment by matkoniecz** [comment]
@westnordost Thanks for the icon!

**Comment by matkoniecz** [comment]
One more thing to do (from testing) - surface=metal is rare but valid value (appearing typically on small footway bridges but sometimes also for some roads - see https://www.openstreetmap.org/way/231559729#map=18/49.77961/22.65244&layers=N). It should appear here (adding it to normal surface quest is probably overkill, especially as adding surface to highway=footway/path is not currently supported).

**Comment by westnordost** [comment]
Nothing happening here, I am closing this.

**Pull request closed without merging**